packages/ember-metal/lib/weak_map.js

```diff
@@ -20,13 +24,19 @@ function UNDEFINED() {}
  * practice, most use cases satisfy this limitation which is why it is included
  * in ember-metal.
  */
-export default function WeakMap() {
-  assert(
-    'Invoking the WeakMap constructor with arguments is not supported at this time',
-    arguments.length === 0
-  );
-
+export default function WeakMap(iterable) {
   this._id = GUID_KEY + (id++);
+
+  if (iterable === null || iterable === undefined) {
+    return;
+  } else if (Array.isArray(iterable)) {
+    for (let i = 0; i < iterable.length; i++) {
+      let entry = iterable[i];
+      this.set(entry[0], entry[1]);
+    }
+  } else {
+    throw new TypeError('Invalid iterable used to initialize weak map');
```

Comment:
Well, es6 Iterables are valid but not supported. I am unsure if this messaging is quite correct.